### PR TITLE
fix(TemplateParser): report empty expression

### LINKED ISSUE #3754: Alpha35: Empty event expression produce cryptic error
Related: #3664 (should be fixed together)

I have a template with the following element:
```html
<input type="file" id="picture" accept="image/*"
                           (change)="onFileSelected(f, $event)"
                           (fileread)>
```
On alpha 33 it worked great.
On alpha35 it produces the following runtime error:
```
EXCEPTION: TypeError: Cannot set property 'lastInBinding' of undefined
angular2.js:13030 STACKTRACE:
angular2.js:13030 TypeError: Cannot set property 'lastInBinding' of undefined
    at Function.execute._ConvertAstIntoProtoRecords.create (http://localhost:5555/lib/angular2.js:18104:47)
    at http://localhost:5555/lib/angular2.js:17730:49
    at Array.map (native)
    at createEventRecords (http://localhost:5555/lib/angular2.js:17729:36)
    at JitProtoChangeDetector.execute.JitProtoChangeDetector._createFactory (http://localhost:5555/lib/angular2.js:17575:39)
    at new JitProtoChangeDetector (http://localhost:5555/lib/angular2.js:17567:32)
    at $__1.execute.JitChangeDetection.createProtoChangeDetector (http://localhost:5555/lib/angular2.js:10323:20)
    at http://localhost:5555/lib/angular2.js:15553:42
    at Array.map (native)
    at Function.execute.ListWrapper.map (http://localhost:5555/lib/angular2.js:6374:26)
```

The following modification still produces the error:
```html
<input type="file" id="picture" accept="image/*"
                           (change)="onFileSelected(f, $event)"
                           (fileread)="">
```

This one works
```html
<input type="file" id="picture" accept="image/*"
                           (change)="onFileSelected(f, $event)"
                           (fileread)="x">
```

### PR DESCRIPTION
**Pull request by vicb**
fixes #3754

### CODE REVIEW

File: modules/@angular/compiler/src/template_parser/template_parser.ts [hunk 1]
```diff
@@ -234,6 +234,12 @@ class TemplateParseVisitor implements html.Visitor {
 
   private _parseAction(value: string, sourceSpan: ParseSourceSpan): ASTWithSource {
     const sourceInfo = sourceSpan.start.toString();
+
+    if (!value || value == '') {
```
Comment (pkozlowski-opensource) [inline modules/@angular/compiler/src/template_parser/template_parser.ts]: What about `(foo)="  "` - I mean - we should also catch whitespaces. More generally I thought that this check needs to be done later, when the expression is parsed and we know if this is a valid action expression or not.

Comment (vicb) [inline modules/@angular/compiler/src/template_parser/template_parser.ts]: Good point, thanks !

**Comment by vicb** [comment]
@pkozlowski-opensource PTAL

**Comment by pkozlowski-opensource** [comment]
lgtm

**Pull request merged**